### PR TITLE
AA-411: Replace "magic" return values with EntryPoint callbacks

### PR DESCRIPTION
**Pull request by forshtat**

### CODE REVIEW

File: RIPS/rip-7560.md [hunk 1]
```diff
@@ -240,25 +235,37 @@ in future revisions of this EIP.
...
-- **magic value** 
-    - `MAGIC_VALUE_SENDER` - This value is returned by the account after it verified the transaction and agreed to pay for its execution.
-    - `MAGIC_VALUE_SIGFAIL` - This value is returned by the account if the transaction is syntactically valid, but the signer is incorrect.
-    - Any other returned value or a reverted execution is considered to be a failure, and the transaction is rejected and not included on-chain.
-- **validUntil** (6 bytes) - a timestamp, or zero for "infinite". The transaction is valid only up to this time.
-- **validAfter** (6 bytes) - a timestamp. The transaction is valid only after this time.
+Calls to the `AA_ENTRY_POINT` approval callbacks have the following meaning:
 
+- `acceptAccount` - This callback is called by the account after it verified the transaction and agreed to pay for its execution.
+- `sigFailAccount` - This callback is called by the account if the transaction is syntactically valid, but the signer is incorrect.
```
Comment (drortirosh) [inline RIPS/rip-7560.md]:
```suggestion
- `sigFailAccount` - This callback is called by the account if the transaction is syntactically valid, but the signature is incorrect.
```

File: RIPS/rip-7560.md [hunk 2]
```diff
@@ -240,25 +235,37 @@ in future revisions of this EIP.
...
+Note that this callback is used during gas estimation and does **not** indicate a valid transaction.
+
+The parameters passed to the callback functions have the following meaning:
+
+- **validAfter** - a timestamp. The transaction is valid only after this time.
+- **validUntil** - a timestamp. The transaction is valid only up to this time. Zero is a special value meaning "valid indefinitely".
+
+Account not making any calls to the `AA_ENTRY_POINT` callbacks, making any call other than the ones defined here,
```
Comment (drortirosh) [inline RIPS/rip-7560.md]: maybe reverse:
the account MUST make exactly one call to the AA_ENTRY_POINT to be considered valid.
calling more times, or not calling at all, calling from a nested contract or reverting is considered a validation failure.

File: RIPS/rip-7560.md [hunk 3]
```diff
@@ -505,13 +529,15 @@ The following is a non-exhaustive list of known security considerations regardin
...
-This EIP adds a new way for a smart contract to have its balance charged simply by returning a valid value from a
+This EIP adds a new way for a smart contract to have its balance charged simply by making a valid callback call
+to the `AA_ENTRY_POINT` address from a
 function with method ID that corresponds to `validateTransaction`, `validatePaymasterTransaction`.
 
 This creates a new kind of risk for contracts that accidentally or maliciously contain such methods but are not public
 about the fact that these contracts can be used as a `sender` or a `paymaster` in an `AA_TX_TYPE` transaction.
 
-This is somewhat mitigated by requiring these contracts to return `MAGIC_VALUE_SENDER` or `MAGIC_VALUE_PAYMASTER`,
+This is somewhat mitigated by requiring these contracts to call `acceptAccount` or `acceptPaymaster` callbacks
```
Comment (drortirosh) [inline RIPS/rip-7560.md]: This _is_ mitigated, not "somewhat".
if a contract could be made to call an arbitrary address with arbitrary calldata, it means that this contract is already vulnerable to token draining.

File: RIPS/rip-7560.md (hunk 2 above)
Comment (forshtat) [reply RIPS/rip-7560.md]: Done.

File: RIPS/rip-7560.md (hunk 3 above)
Comment (forshtat) [reply RIPS/rip-7560.md]: Rephrased, removed "somewhat".

**Comment by forshtat** [comment]
This PR seems ready, opened in the Ethereum repository:
https://github.com/ethereum/RIPs/pull/33

**Pull request closed without merging**